modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -141,12 +142,17 @@ private CompletableFuture<?> onIndexBuilding(StartBuildingIndexEventParameters p
 
             var startBuildIndexFutures = new ArrayList<CompletableFuture<?>>();
 
-            for (TablePartitionId primaryReplicaId : primaryReplicaIds) {
-                if (primaryReplicaId.tableId() == indexDescriptor.tableId()) {
-                    CompletableFuture<?> startBuildIndexFuture = getMvTableStorageFuture(parameters.causalityToken(), primaryReplicaId)
-                            .thenCompose(mvTableStorage -> awaitPrimaryReplica(primaryReplicaId, clockService.now())
+            for (ZonePartitionId zonePartitionId : primaryReplicaIds) {
+
+                int tableId = zonePartitionId.tableId();
+
+                TablePartitionId tablePartId = new TablePartitionId(tableId, zonePartitionId.partitionId());
+
+                if (tableId == indexDescriptor.tableId()) {
+                    CompletableFuture<?> startBuildIndexFuture = getMvTableStorageFuture(parameters.causalityToken(), tablePartId)
+                            .thenCompose(mvTableStorage -> awaitPrimaryReplica(zonePartitionId, clockService.now())
                                     .thenAccept(replicaMeta -> tryScheduleBuildIndex(
-                                            primaryReplicaId,
+                                            tablePartId,
```

Comment:
but how the we will rebuild indexes without tableId? I believe that it can be done only when we have one partition for all tables within a zone, so index also will be one for a pair (zoneId, partId)